SwiftUIFourierSeries/Views/ConfigView.swift

```diff
@@ -3,46 +3,52 @@ import SwiftUI
 
 struct ConfigView: View {
     @ObservedObject var store: Store
+    @Environment(\.verticalSizeClass) var verticalSizeClass: UserInterfaceSizeClass?
+    @Environment(\.horizontalSizeClass) var horizontalSizeClass: UserInterfaceSizeClass?
     
     var body: some View {
-        HStack {
-            VStack {
-                HStack {
-                    if self.store.state.timerIsRunning {
-                        Button("Pause") { self.store.send(.pause) }
-                    } else {
-                        Button("Play") { self.store.send(.play) }
+        // https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/
+        Group {
+            if verticalSizeClass == .regular && horizontalSizeClass == .compact {
+                // iPhone Portrait or iPad 1/3 split view for Multitasking for instance
+                // A
+                // B
+                // C
+                VStack {
+                    Spacer()
+                    VStack {
```

Comment:
I am not a SwiftUI expert, so I apologise if I missed something.
I think there are a few things we can optimise here:

- Most (if not all) `Spacer` can be removed. They don't seem to affect the layout.
- That makes wrapping a stack in a stack unnecessary.
- Distinguishing between standard and larger phones doesn't seem to do much.
- On iPads, both size classes are `regular`. This case is not handled here.

I'd propose to condense the layout code into two cases:
- iPhone portrait and iPad with 1/3 split (with a vertical layout)
- Everything else (with a horizontal layout)

In the end, this could look something like this:

```
var body: some View {
        // https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/
        if verticalSizeClass == .regular && horizontalSizeClass == .compact {
            // iPhone Portrait or iPad 1/3 split view for Multitasking for instance
            // A
            // B
            // C
            VStack {
                ButtonsView(store: store).padding()
                PickerView(store: store).padding()
            }
        } else {
            // Landscape layout for all other size class combinations
            // A B C
            HStack {
                ButtonsView(store: store).padding()
                PickerView(store: store).padding()
            }
        }
    }
```

What do you think?

Reply:
I'll get right on it, SwiftUI is fairly new to me as well so thank you for the feedback, Simon!

Reply:
Thank you!